### PR TITLE
Enable turning on/off vehicle headlights with environmental brightness.

### PR DESCRIPTION
**Pull request by gio-tempo**
This is done by syncing the Brightness value from an `ATempoBrightnessMeter` Actor that gets automatically spawned in the level to Mass via `UTempoMassAgentBrightnessMeterSyncTrait`.  The user can place an `ATempoBrightnessMeter` Actor in the level themselves to override its placement, and the placed `ATempoBrightnessMeter` Actor will be used instead.

ISM representations will always have their headlights off for now.

Note:  This was done in support of enabling vehicle colors and vehicle lights.  Therefore, this change also includes a fix to allow Mass Traffic to generate a new seed (to allow randomization between runs by default) as advertised when `UMassTrafficSettings::RandomSeed` is set to 0.

### CODE REVIEW

File: External/Traffic/Source/MassTraffic/Public/MassTrafficSettings.h [hunk 1]
```diff
@@ -319,6 +315,18 @@ class MASSTRAFFIC_API UMassTrafficSettings : public UMassModuleSettings
 	UPROPERTY(EditAnywhere, Config, Category="Obstacle Avoidance")
 	FVector2D MinimumDistanceToObstacleRange = { 80.0f, 300.0f };
 
+	/** Maximum vehicle speed that can be stopped quickly (MpH.) */ 
+	UPROPERTY(EditAnywhere, Config, Category="Vehicles")
+	float MaxQuickStopSpeedMPH = 5.0f;
+
+	// Environmental brightness threshold [0..1] at which vehicles will turn *on* their headlights.
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Public/MassTrafficSettings.h:322]:
```suggestion
	// Environmental brightness threshold [0..1] below which vehicles will turn *on* their headlights.
```

File: External/Traffic/Source/MassTraffic/Public/MassTrafficSettings.h [hunk 2]
```diff
@@ -319,6 +315,18 @@ class MASSTRAFFIC_API UMassTrafficSettings : public UMassModuleSettings
...
+	UPROPERTY(EditAnywhere, Config, Category="Vehicles")
+	float VehicleTurnOnHeadlightsBrightnessThreshold = 0.4f;
+
+	// Environmental brightness threshold [0..1] at which vehicles will turn *off* their headlights.
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Public/MassTrafficSettings.h:326]:
```suggestion
	// Environmental brightness threshold [0..1] above which vehicles will turn *off* their headlights.
```

File: External/Traffic/Source/MassTraffic/Private/MassTrafficVehicleVisualizationProcessor.cpp [hunk 3]
```diff
@@ -242,12 +244,27 @@ void UMassTrafficVehicleUpdateCustomVisualizationProcessor::Execute(FMassEntityM
...
+			FMassTrafficVehicleLightsFragment& VehicleStateFragment = VehicleStateFragments[EntityIdx];
 			FMassActorFragment& ActorFragment = ActorFragments[EntityIdx];
 			FMassRepresentationFragment& RepresentationFragment = VisualizationFragments[EntityIdx];
 
+			if (!EnvironmentalBrightnessFragments.IsEmpty())
+			{
+				const FEnvironmentalBrightnessFragment& EnvironmentalBrightnessFragment = EnvironmentalBrightnessFragments[EntityIdx];
+
+				// Automatically turn headlights on and off with the vehicle's current level of environmental brightness.
+				VehicleStateFragment.bHeadlights = (!VehicleStateFragment.bHeadlights && EnvironmentalBrightnessFragment.Brightness <= MassTrafficSettings->VehicleTurnOnHeadlightsBrightnessThreshold) ||
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Private/MassTrafficVehicleVisualizationProcessor.cpp:258]: Is the "or equal to" important here? (maybe it was before you separated the turn on and off thresholds?) If not, let's remove it?

**Review by pete-tempo — APPROVED** [review]
Approved with minor comments

File: External/Traffic/Source/MassTraffic/Private/MassTrafficVehicleVisualizationProcessor.cpp (hunk 3 above)
Comment (gio-tempo) [reply External/Traffic/Source/MassTraffic/Private/MassTrafficVehicleVisualizationProcessor.cpp:258]: I always had the 2 thresholds.  But, the "or equal to" isn't really important.  It's just meant to give symmetry to the `off -> on` and `on -> off` transitions.

The `on -> off` transition happens when `EnvironmentalBrightnessFragment.Brightness < MassTrafficSettings->VehicleTurnOffHeadlightsBrightnessThreshold` is false.  Or, in other words via an "implied" `EnvironmentalBrightnessFragment.Brightness >= MassTrafficSettings->VehicleTurnOffHeadlightsBrightnessThreshold`, which includes the threshold value itself.

**Review by pete-tempo — APPROVED** [review]

**Pull request merged**